src/deluge/gui/ui/rename/rename_midi_cc_ui.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright © 2019-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "gui/ui/rename/rename_midi_cc_ui.h"
+#include "definitions_cxx.hpp"
+#include "extern.h"
+#include "gui/l10n/l10n.h"
+#include "gui/views/automation_view.h"
+#include "hid/buttons.h"
+#include "hid/display/display.h"
+#include "hid/led/pad_leds.h"
+#include "model/instrument/midi_instrument.h"
+#include "model/output.h"
+#include "model/song/song.h"
+
+RenameMidiCCUI renameMidiCCUI{};
+
+RenameMidiCCUI::RenameMidiCCUI() {
+	title = "CC Name";
+}
+
+bool RenameMidiCCUI::opened() {
+	bool success = QwertyUI::opened();
+	if (!success) {
+		return false;
+	}
+
+	Clip* clip = getCurrentClip();
+
+	int32_t cc = clip->lastSelectedParamID;
+
+	// if we're not dealing with a real cc number
+	// then don't allow user to edit the name
+	if (cc < 0 || cc >= kNumRealCCNumbers) {
```

Comment:
CC1 needs to be blocked too (it's mod wheel and stored internally as cc 130 or something)